File: maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double curr = 0;
        
        
        for (int right = 0; right < k; right ++){
            curr += nums[right];
        }
        double ans = curr;
        for (int i = k; i < nums.length; i ++){
            curr += nums[i] - nums[i-k];
            
            ans = Math.max(ans, curr);
        }
        
        return ans /k;
        
    }
}

File: maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left = ans = curr = 0
        
        for i in range(k):
            curr += nums[i]
            
        ans = curr/k 
        
        for right in range(k, len(nums)):
            curr +=  nums[right] - nums[right - k ] 
            ans = max(ans, curr/k)
            
        return ans
            
            
        